tests/test_api_evolution_traversal.py

```
@@ -0,0 +1,57 @@
+
+import pytest
+from fastapi import HTTPException
+from swarm.api.routes import evolution
+
+
+@pytest.mark.anyio
+async def test_evolution_path_traversal():
+    """Test that evolution endpoints validate run_id against path traversal."""
+
+    # Mock runs root
+    runs_root = Path("/tmp/runs")
```

Ruff (0.14.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/runs"

(S108)